Buildings/Controls/OBC/CDL/Logical/Sources/TimeTable.mo

```diff
@@ -0,0 +1,235 @@
+    annotation (Placement(transformation(extent={{-80,70},{-60,90}})));
+  Integers.Sources.Constant intTru[nout](k=fill(1, nout)) "Integer true"
+    annotation (Placement(transformation(extent={{-80,40},{-60,60}})));
+  Or or2[nout] "Checks if the table value is either true or false"
+    annotation (Placement(transformation(extent={{20,60},{40,80}})));
+  MultiAnd mulAnd(
+    final nu=nout) "Multi and"
+    annotation (Placement(transformation(extent={{50,60},{70,80}})));
+  MultiOr mulOr1(
+    final nu=nout) "Multi or"
+    annotation (Placement(transformation(extent={{20,-100},{40,-80}})));
+  And and2
+    "Logical and"
+    annotation (Placement(transformation(extent={{60,20},{80,40}})));
+  Buildings.Controls.OBC.CDL.Conversions.RealToInteger realToInteger[nout]
+    "Converts scheduled values to integer"
+    annotation (Placement(transformation(extent={{-60,-10},{-40,10}})));
+  // CDL uses different enumerations for extrapolation
+  // than the Modelica Standard Library. Hence, we cast the CDL
+  // enumeration to the MSL enumeration.
+  Modelica.Blocks.Sources.CombiTimeTable tab(
+    final tableOnFile=false,
+    final table=table,
+    final columns=2:size(tab.table, 2),
+    final smoothness=Modelica.Blocks.Types.Smoothness.ConstantSegments,
+    final extrapolation=if extrapolation == CDL.Types.Extrapolation.HoldLastPoint then
+                          Modelica.Blocks.Types.Extrapolation.HoldLastPoint
+                        else
+                          Modelica.Blocks.Types.Extrapolation.Periodic,
+    final startTime=if (extrapolation == Types.Extrapolation.Periodic) then integer(t0/86400)*86400 else 0,
+    final timeScale=timeScale) "Time table"
+    annotation (Placement(transformation(extent={{-100,-10},{-80,10}})));
+
+  Utilities.Assert assMes1(message="Scheduled values are not all zeroes or ones")
+    "Assert all scheduled values are either zero or one values"
+    annotation (Placement(transformation(extent={{100,60},{120,80}})));
+  Conversions.IntegerToReal intToRea[nout] "Type conversion"
+    annotation (Placement(transformation(extent={{-20,-30},{0,-10}})));
+  Continuous.Add add2[nout](final k1=fill(1, nout), final k2=fill(-1, nout))
+    "Subtracts inputs from their integer conversion results"
+    annotation (Placement(transformation(extent={{20,-60},{40,-40}})));
+  Continuous.Abs abs1[nout]
+    "Absolute value"
+    annotation (Placement(transformation(extent={{-60,-100},{-40,-80}})));
+  Continuous.GreaterThreshold greThr[nout](
+    final t=fill(Constants.small, nout))
+    "Value comparisson"
+    annotation (Placement(transformation(extent={{-20,-100},{0,-80}})));
+  Not not1
+    "Logical not"
+    annotation (Placement(transformation(extent={{60,-100},{80,-80}})));
+
+initial equation
+  t0=time;
+
+equation
+  assert(n > 0, "No table values defined.");
+  assert(extrapolation <> CDL.Types.Extrapolation.LastTwoPoints, "Unsuitable extrapolation setting.");
```

Comment:
Move to initial equation section

Reply:
Sure, and I removed the second line as we now only use periodic. This applies to both integer and boolean flavors of the block.